File: google_kickstarter/2016/diwali_lightings.py
def main():
    test_cases = get_test_cases()

    blue_count = count_number_of_blues(test_cases)

    display_test_case_results(blue_count)


def get_test_cases():

    T = int(input())
    test_cases = []

    for i in range(T):
        t_case = {}
        t_case['S'] = input()
        indices = input()
        [t_case["i"], t_case['j']] = list(map(int, indices.split(" ")))
        test_cases.append(t_case)

    return test_cases


def count_number_of_blues(test_cases):
    blue_count = []
    count = 0
    s = ''
    i = j = 0

    for test in test_cases:
        [s, i, j] = [test['S'], test['i'], test['j']]
        s = s*(j//len(s))
        s = s[i-1:j]
        count = s.count('B')
        blue_count.append(count)

    return blue_count


def display_test_case_results(blue_count):
    for i in range(len(blue_count)):
        print(f"Case #{i + 1}: {blue_count[i]}")


if __name__ == "__main__":
    main()

File: google_kickstarter/2016/diwali_lightings.dart
import 'dart:io';

void main() {
  //LOGGING: print("start of main method");
  List<Map<String, dynamic>> testCases = getTestCases();

  List<int> blueCount = countNumberOfBlues(testCases);

  displayTestCaseResults(blueCount);
  //LOGGING: print("end of main method");
}

List<Map<String, dynamic>> getTestCases() {
  //LOGGING: print("start of user input");
  dynamic userInput = stdin.readLineSync();
  int T = 0;

  T = int.parse(userInput);

  List<Map<String, dynamic>> testCases = [];
  Map<String, dynamic> tCase = {};
  for (var i = 0; i < T; i++) {
    tCase = {};
    userInput = stdin.readLineSync();

    tCase['s'] = userInput;

    userInput = stdin.readLineSync();

    userInput = userInput.split(" ").map((e) => int.parse(e)).toList();
    tCase["i"] = userInput[0];
    tCase['j'] = userInput[1];
    testCases.add(tCase);
  }
  //LOGGING: print("end of user input");

  return testCases;
}

List<int> countNumberOfBlues(testCases) {
  List<int> blueCount = [];
  int count = 0, i = 0, j = 0;
  String s = '';

  //LOGGING: print("count no. of blues begins");
  for (var test in testCases) {
    //LOGGING: print("begin count loop");
    s = test['s'];
    i = test['i'];
    j = test['j'];
    s = s * (j ~/ s.length);
    s = s.substring(i - 1, j);
    count = 'B'.allMatches(s).length;
    blueCount.add(count);
    //LOGGING: print("end count loop");
  }
  //LOGGING: print("count no. of blues ends");
  return blueCount;
}

void displayTestCaseResults(List<int> blueCount) {
  //LOGGING: print("started displaying results");
  for (var i = 0; i < blueCount.length; i++) {
    stdout.write("Case #${i + 1}: ${blueCount[i]}\n");
  }
  //LOGGING: print("done displaying results");
}
